File: usr/lib/node/atomicpi.js

const sig = (chip, chipIndex, globalIndex) => ({
  chip: `gpiochip${chip}`,
  chip_idx: chipIndex,
  global_idx: globalIndex
})

const signals = {
  // External GPIO
  ISH_GPIO_0: sig(3, 21, 335),
  ISH_GPIO_1: sig(3, 18, 332),
  ISH_GPIO_2: sig(3, 24, 338),
  ISH_GPIO_3: sig(3, 15, 329),
  ISH_GPIO_4: sig(3, 22, 336),
  ISH_GPIO_7: sig(3, 16, 330),
  // Volume Up Pin
  GPIO_DFX_2: sig(1, 7, 348),
  // Volume Down Pin
  GPIO_DFX_4: sig(1, 5, 346),
  // Internal Signals
  I2C2_3P3_SDA: sig(0, 62, 476),
  I2C2_3P3_SCL: sig(0, 66, 480),
  AU_MIC_SEL: sig(1, 0, 341),
  XMOS_RESET: sig(1, 8, 349),
  BN_INT: sig(1, 17, 358),
  BN_RESET: sig(1, 25, 366)
}

// 26-pin connector descriptions
const atomicpi = {
  index_of: {
    ISH_GPIO_0: 24,
    ISH_GPIO_1: 25,
    ISH_GPIO_2: 26,
    ISH_GPIO_3: 18,
    ISH_GPIO_4: 19,
    ISH_GPIO_7: 20
  }
}

// Enchilada Breakout Board
const enchilada = {
  index_of: {
    ISH_GPIO_0: 9,
    ISH_GPIO_1: 10,
    ISH_GPIO_2: 11,
    ISH_GPIO_3: 3,
    ISH_GPIO_4: 4,
    ISH_GPIO_7: 5
  },
  leds: {
    green: 'ISH_GPIO_1',
    yellow: 'ISH_GPIO_2'
  }
}

// Physical Connector Descriptions
const connectors = {
  atomicpi,
  enchilada
}

Object.assign(module.exports, {
  /* Kernel-visible signals */
  signals,
  connectors
})

File: usr/lib/python2.7/dist-packages/atomicpi.py
__all__ = ["signals", "atomicpi", "enchilada"]

class _data(dict):
  def __getattr__(self, name):
    return self[name]

# Kernel-visible signals
signals = _data(
  #  External GPIO
    ISH_GPIO_0    = _data(chip='gpiochip3', chip_idx=21, global_idx=335),
    ISH_GPIO_1    = _data(chip='gpiochip3', chip_idx=18, global_idx=332),
    ISH_GPIO_2    = _data(chip='gpiochip3', chip_idx=24, global_idx=338),
    ISH_GPIO_3    = _data(chip='gpiochip3', chip_idx=15, global_idx=329),
    ISH_GPIO_4    = _data(chip='gpiochip3', chip_idx=22, global_idx=336),
    ISH_GPIO_5    = _data(chip='gpiochip3', chip_idx=16, global_idx=330),
  # Volume up pin
    GPIO_DFX_2    = _data(chip='gpiochip1', chip_idx=7, global_idx=348),
  # Volume down pin
    GPIO_DFX_4    = _data(chip='gpiochip1', chip_idx=5, global_idx=346),
  # Internal signals
    I2C2_3P3_SDA  = _data(chip='gpiochip0', chip_idx=62, global_idx=476),
    I2C2_3P3_SCL  = _data(chip='gpiochip0', chip_idx=66, global_idx=480),
    AU_MIC_SEL    = _data(chip='gpiochip1', chip_idx=0,  global_idx=341),
    XMOS_RESET    = _data(chip='gpiochip1', chip_idx=8,  global_idx=349),
    BN_INT        = _data(chip='gpiochip1', chip_idx=17, global_idx=358),
    BN_RESET      = _data(chip='gpiochip1', chip_idx=25, global_idx=366)
)

# Physical connector descriptions
connectors = _data(
  # 26 pin connector interface
  atomicpi = _data(
    index_of = _data(
      ISH_GPIO_0 = 24,
      ISH_GPIO_1 = 25,
      ISH_GPIO_2 = 26,
      ISH_GPIO_3 = 18,
      ISH_GPIO_4 = 19,
      ISH_GPIO_7 = 20
    )
  ),
  enchilada = _data(
    index_of = _data(
      ISH_GPIO_0 = 9,
      ISH_GPIO_1 = 10,
      ISH_GPIO_2 = 11,
      ISH_GPIO_3 = 3,
      ISH_GPIO_4 = 4,
      ISH_GPIO_7 = 5
    ),
    leds = _data(
      green = "ISH_GPIO_1",
      yellow = "ISH_GPIO_2"
    )
  )
)